util/PythonCommon.h

```diff
@@ -0,0 +1,50 @@
+#ifndef __FreeOrion__Util__PythonCommon__
+#define __FreeOrion__Util__PythonCommon__
+
+#include <boost/python.hpp>
+
+#include "Export.h"
+
+class FO_COMMON_API PythonCommon {
+public:
+    PythonCommon();
+    virtual ~PythonCommon();
+
+    /**
+       IsPythonRunning returns true is the python interpreter is
+       initialized.  It is typically called after
+       HandleErrorAlreadySet() to determine if the error caused the
+       interpreter to shutdown.
+     */
+    bool         IsPythonRunning() const;
+
+    /**
+       Handles boost::python::error_already_set.
+       If the error is SystemExit the python interpreter is finalized
+       and no longer available.
+
+       Call PyErr_Print() if the exception is an error.
+
+       HandleErrorAlreadySet is idempotent, calling it multiple times
+       won't crash or hang the process.
+     */
+    void HandleErrorAlreadySet();
+
+    bool         Initialize();        // initializes and runs the Python interpreter, prepares the Python environment
+
+    virtual bool InitCommonImports(); // initializes Python imports, should be implemented by derived classes
```

Comment:
> should be implemented by derived classes

Contract via comment does not always work.  Maybe you can make it abstract and make the compiler care about overriding.  Or put text inside the default implementation as an exception, then you will be warned in runtime.

Reply:
> make it abstract

pure virtual function `= 0` would make sense

Reply:
Client and test doesn't override it because don't use python imports. I'll write comment differently.